state/accumulator.go

```diff
@@ -494,20 +507,27 @@ func (a *Accumulator) Accumulate(txn *sqlx.Tx, userID, roomID string, prevBatch
 			snapID = newSnapshot.SnapshotID
 		}
 		if err := a.eventsTable.UpdateBeforeSnapshotID(txn, ev.NID, beforeSnapID, replacesNID); err != nil {
-			return 0, nil, err
+			return AccumulateResult{}, err
 		}
 	}
 
+	if len(redactTheseEventIDs) > 0 {
+		// TODO: this is going to send out a cache invalidation when someone redacts a
+		// message event, which is unnecessary. Can we get eventsTable.Redact to tell us
+		// if it redacted any state events, and only include a snapID here if so?
+		result.RequiresReload = true
```

Comment:
We probably need to do this before landing this PR as otherwise redactions will be expensive.

Reply:
Have done this. Added a unit test to keep me honest, too.